packages/livechat/src/routes/Register/index.tsx

```diff
@@ -0,0 +1,305 @@
+import { route } from 'preact-router';
+import { useContext, useEffect } from 'preact/hooks';
+import type { Control, FieldErrors, FieldValues, SubmitHandler } from 'react-hook-form';
+import { Controller, useForm } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+
+import { Livechat } from '../../api';
+import { Button } from '../../components/Button';
+import { Form, FormField, TextInput, SelectInput } from '../../components/Form';
+import Screen from '../../components/Screen';
+import { createClassName, sortArrayByColumn } from '../../components/helpers';
+import CustomFields from '../../lib/customFields';
+import { validateEmail } from '../../lib/email';
+import { parentCall } from '../../lib/parentCall';
+import { StoreContext } from '../../store';
+import styles from './styles.scss';
+
+// TODO: Properly type departments in livechat
+type department = {
+	_id: string;
+	name: string;
+	[key: string]: unknown;
+};
+
+type CustomField = {
+	_id: string;
+	required?: boolean;
+	label?: string;
+	type: string;
+	options?: string[];
+	defaultValue?: string;
+	regexp?: RegExp;
+};
+
+type RenderCustomFieldsProps = {
+	customFields: CustomField[];
+	loading: boolean;
+	control: Control;
+	errors: FieldErrors<FieldValues>;
+};
+
+const renderCustomFields = ({ customFields, loading, control, errors }: RenderCustomFieldsProps) =>
+	customFields.map(({ _id, required = false, label, type, options, regexp, defaultValue }) => {
+		const { t } = useTranslation();
+		switch (type) {
+			case 'input':
+				return (
+					<FormField label={label} required={required} key={_id} error={errors?.[_id]?.message?.toString()}>
+						<Controller
+							name={_id}
+							control={control}
+							defaultValue={defaultValue}
+							rules={{
+								required,
+								...(regexp && {
+									pattern: {
+										value: regexp,
+										message: t('invalid', { field: label }),
+									},
+								}),
+							}}
+							render={({ field }) => (
+								<TextInput name={_id} placeholder={t('insert_your_field_here', { field: label })} disabled={loading} field={field} />
+							)}
+						/>
+					</FormField>
+				);
+			// TODO: typeguards for different types of custom fields
+			case 'select':
+				return (
+					<FormField label={label} required={required} key={_id} error={errors?.[_id]?.message?.toString()}>
+						<Controller
+							name={_id}
+							control={control}
+							rules={{ required }}
+							defaultValue={defaultValue}
+							render={({ field }) => (
+								<SelectInput
+									name={_id}
+									placeholder={t('choose_an_option')}
+									options={(options as string[])?.map((option: string) => ({ value: option, label: option }))}
+									disabled={loading}
+									field={field}
+								/>
+							)}
+						/>
+					</FormField>
+				);
+		}
+		return null;
+	});
```

Comment:
This can be its own component `<CustomFields />`. We should also re-use the `Controller` for both field types.